### PR TITLE
Dra 517 preservicaid workflow

### PR DESCRIPTION
**Pull request by thomasegense**
New API method: fetchKalturaIdsAndUpdateRecords
1)Method will batch over record to get referenceId for records.
2) For batch call preservica (single call) to get the referenceId<->kalturaId mapping
3) Update the mapping table for each batch.
4) When all batches are completed, it will enrich records with kalturaID from the mapping table.

Notice there was some manual merge issues from master I had to fix

### CODE REVIEW

File: src/main/java/dk/kb/datahandler/facade/DsDatahandlerFacade.java [hunk 1]
```diff
@@ -49,9 +54,80 @@
...
-    
+
     private static DsStorageClient storageClient;
-    
+
+    /**
+     * <p>
+     * Will load all referenceId defined in storage and call Kaltura to map them to KalturaId.
+     * The results will be saved in ds-storage in the mapping table.
+     * When all mappings has been updated, the records will enriched with the kalturaId.
```
Comment (VictorHarbo) [inline src/main/java/dk/kb/datahandler/facade/DsDatahandlerFacade.java:64]: will be

File: src/main/java/dk/kb/datahandler/facade/DsDatahandlerFacade.java [hunk 2]
```diff
@@ -49,9 +54,80 @@
...
+     * If the job fails alll mappings are loaded the records will not be enriched.
```
Comment (VictorHarbo) [inline src/main/java/dk/kb/datahandler/facade/DsDatahandlerFacade.java:65]: Is this sentence missing an 'and'? ... and the records will not...

**Review by VictorHarbo — CHANGES_REQUESTED** [review]
Found a weird unused import. Remove that one and we are good to go!

File: src/main/java/dk/kb/datahandler/facade/DsDatahandlerFacade.java [hunk 3]
```diff
@@ -20,6 +21,7 @@
 import dk.kb.util.webservice.stream.ContinuationInputStream;
 import org.apache.commons.io.IOUtils;
 import org.apache.solr.client.solrj.SolrServerException;
+import org.checkerframework.checker.units.qual.m;
```
[the remark this answers is not in the document]
Comment (thomasegense) [reply src/main/java/dk/kb/datahandler/facade/DsDatahandlerFacade.java:24]: Wtf is that.

File: src/main/java/dk/kb/datahandler/facade/DsDatahandlerFacade.java (hunk 1 above)
Comment (thomasegense) [reply src/main/java/dk/kb/datahandler/facade/DsDatahandlerFacade.java:64]: fixed

File: src/main/java/dk/kb/datahandler/facade/DsDatahandlerFacade.java (hunk 2 above)
Comment (thomasegense) [reply src/main/java/dk/kb/datahandler/facade/DsDatahandlerFacade.java:65]: rewritten

**Review by VictorHarbo — APPROVED** [review]
Super!

**Pull request merged**